### ISSUE TITLE
Buggy Android 1.0.0 Version

### ISSUE THREAD

**Issue by HarshKapadia2** [TryQuiet/quiet#1391]
Hey, I am an Android user of Quiet.

I installed Quiet about a week ago, but had to uninstall it within a few hours. I reinstalled it a few hours ago, but I am still facing the same issues. I'll list some of them below.

Some of the issues I faced:
- Chats don't open, even on repeated clicking, after the app is opened the second time after installing it. The first use as soon as one gains access to the chats is fine, but all the issues start after that.
- Once a chat opens somehow, I cannot click on the chat input. Again, the app responds after a LONG LONG time and I don't even know when it does that. I just let it run in the background and then open it to check if the input has responded.
- I have the same issue with the send button after typing the message.
- The app crashes sometimes as well.
- Is the 'Quiet backend running' non-dismissable notification really required for a user? It just occupies space in my notification bar.

Essentially, I just cannot use the app. I cannot send messages and can barely read messages.

Mobile specs:
- Quiet version: 1.0.0 (as displayed on the app start screen)
- Mobile model: Samsung Galaxy M30
- Android version: 10
- One UI Code: 2.0
- Kernel version: 4.4.177-22392479

Let me know if you need any more information. Thank you!

**Comment by holmesworcester** [comment]
Sorry you ran into all of these issues! Gah! And thank you for the detailed write up.

Two things we haven't done much of yet are testing on a range of android devices and optimizing the initial onboarding experience. Since we've been able to use Quiet ourselves as our daily driver in the community I invited you to, my guess is that the problem lies in one of these two areas.

Quiet does a lot more work when you first join a community and this can slow down the app. It could be you're getting stuck in this for a while or that the symptoms are worse on your device than they are on our other devices. Or it could be that we just ignored the slowness until it caught up and began working. We're focusing now on making onboarding and joining communities work better and based on your experience it seems that we need to focus on the performance piece!

Another possibility is that there's some general problem with quiet on your device that has nothing to do with onboarding at all. I'll make sure the team has this same device to test on and we'll go from there.

For the 'backend is running' item in the notification bar, that used to be required but isn't anymore, so we can make it removable. I'll create an issue for that.

**Comment by holmesworcester** [comment]
Two questions:

1. I see that the M30 has a 4GB and 6GB model. Which one do you have?

2. If you have time and are interested in pursuing this more, could you run the test app provided at https://dontkillmyapp.com/? Samsung ships features that affect any apps running in the background and I'm wondering if that is the cause here.

Thanks so much and again I'm sorry we ran into problems on *both* your computer and phone. 🤦 That's super embarrassing and we'll try to fix it soon!

**Comment by HarshKapadia2** [comment]
> I see that the M30 has a 4GB and 6GB model. Which one do you have?

I have the 4 GB RAM model. None of the other apps on my phone face any issues with a 4 GB RAM though, at least none that I have noticed.

> If you have time and are interested in pursuing this more, could you run the test app provided at [dontkillmyapp.com](https://dontkillmyapp.com/) and share the results?

I'll try to run it, but the tests take a lot of time to run during which I cannot use my mobile phone, so I'm not sure. I'll try to run the 1 hour version.

**Comment by holmesworcester** [comment]
> I have the 4 GB RAM model. None of the other apps on my phone face any issues with a 4 GB RAM though, at least none that I have noticed.

Okay, we'll get our hands on one of these and hopefully we can reproduce the problem.

> I'll try to run it, but the tests take a lot of time to run during which I cannot use my mobile phone, so I'm not sure. I'll try to run the 1 hour version.

Oh, I forgot that these tests take over one's phone. We should be able to get one of these phones soon and run the tests ourselves.

**Comment by holmesworcester** [comment]
> Is the 'Quiet backend running' non-dismissable notification really required for a user? It just occupies space in my notification bar.

For this, I created the following issue: https://github.com/TryQuiet/quiet/issues/1395

However, Android before version 12 does not allow users to dismiss this kind of notification. It's just something that shows whenever an application is running a "foreground service" (a confusing name for whenever something is running constantly in the background). So we can fix this but if you're on Android 10 the fix won't affect you.

**Comment by holmesworcester** [comment]
Moving to "blocked" while we wait for one of these devices to arrive.

**Comment by holmesworcester** [comment]
Just adding a note here that Quiet eventually worked on this device but it kept crashing and was generally difficult to get through the initial sync phase.

**Comment by HarshKapadia2** [comment]
> Just adding a note here that Quiet eventually worked on this device but it kept crashing and was generally difficult to get through the initial sync phase.

Yup. I had to open the app and wait for a long time for chats to get updates. Once that was done, I clicked on a chat and waited a long time for it to open up. Once it opened up, I clicked on the input field, which surprisingly popped up quick. The 'send' button also worked in one go this time, which was not the case in the two Quiet app installs that I had previously done on that mobile device.

**Comment by holmesworcester** [comment]
Just adding that in retrospect this looks like a performance problem. I don't think we're blocked on addressing this. Moving to mobile backlog.
